extensions/amp-date-picker/0.1/amp-date-picker.js

```diff
@@ -337,20 +336,39 @@ class AmpDatePicker extends AMP.BaseElement {
 
     const locale = this.element.getAttribute('locale') || DEFAULT_LOCALE;
     this.moment_.locale(locale);
+
+    /** @private @const */
+    this.props_ = this.getProps_();
+
+    /** @private @const */
+    this.state_ = this.getInitialState_();
   }
 
   /** @override */
   isLayoutSupported(layout) {
-    // TODO(cvializ): Use layout fixed and then expand?
-    return layout == Layout.CONTAINER;
+    return this.mode_ == DatePickerMode.STATIC ?
+      isLayoutSizeDefined(layout) :
+      Layout.CONTAINER;
   }
 
   /** @override */
   layoutCallback() {
     this.setupTemplates_();
     this.setupListeners_();
-    // Make sure it's rendered and measured properly
-    return this.render(this.state_);
+    // Make sure it's rendered and measured properly. Then if possible, attempt
+    // to adjust expand the height to fit the element for static pickers.
+    return this.render(this.state_).then(() => {
+      if (this.mode_ == DatePickerMode.STATIC) {
+        this.vsync_.measure(() => {
+          const scrollHeight = this.container_./*OK*/scrollHeight;
+          const height = this.element./*OK*/offsetHeight;
+          if (scrollHeight > height) {
+            // Add 1px to allow the bottom border to show
+            this.attemptChangeHeight(scrollHeight + 1).catch(() => {});
```

Comment:
I think this needs to become changeHeight instead of attempt. A UI that needs to provide a "show more" button just to get a calendar become fully visible is not great. Given we already force change for way more non-specialized components like twitter, it will be fine for date picker